src/uct/ib/dc/dc_mlx5_ep.h

```diff
@@ -352,11 +352,14 @@ uct_dc_mlx5_iface_progress_pending(uct_dc_mlx5_iface_t *iface,
                           uct_dc_mlx5_iface_dci_can_alloc(iface, pool_index)));
 }
 
-static inline int uct_dc_mlx5_iface_dci_ep_can_send(uct_dc_mlx5_ep_t *ep)
+static inline int uct_dc_mlx5_iface_dci_ep_can_send(uct_dc_mlx5_ep_t *ep,
+                                                    int flush_cancel)
 {
     uct_dc_mlx5_iface_t *iface = ucs_derived_of(ep->super.super.iface, uct_dc_mlx5_iface_t);
     return (!(ep->flags & UCT_DC_MLX5_EP_FLAG_TX_WAIT)) &&
-           uct_rc_fc_has_resources(&iface->super.super, &ep->fc) &&
+           (/* Ignore FC limitations when performing flush(CANCEL) */
```

Comment:
can we ignore `ep->flags & UCT_DC_MLX5_EP_FLAG_TX_WAIT` for flush cancel as well?
Then we could just check for `uct_dc_mlx5_iface_dci_has_tx_resources(iface, ep->dci)` in flush instead of modifying this function

Reply:
good catch, update the functuon.
can't get rid of calling this function in DC's `flush()` we still should check `ep->flags & UCT_DC_MLX5_EP_FLAG_TX_WAIT` and `uct_dc_mlx5_iface_dci_has_tx_resources(iface, ep->dci)` in case of `flush(LOCAL)`.